api/src/main/java/com/cloud/storage/MigrationOptions.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.cloud.storage;
+
+import java.io.Serializable;
+
+public class MigrationOptions implements Serializable {
+
+    private String srcPoolUuid;
+    private Storage.StoragePoolType srcPoolType;
+    private Type type;
+    private String srcBackingFilePath;
+    private boolean copySrcTemplate;
+    private String srcVolumeUuid;
+    private int timeout;
+
```

Comment:
The object state is different for Clone types: LinkedClone (**srcPoolUuid**, **srcPoolType**, _srcBackingFilePath_, _copySrcTemplate_) and FullClone (**srcPoolUuid**, **srcPoolType**, _srcVolumeUuid_). May be better class design here?

Reply:
Well the idea of the class is to capture migration options, I don't see why two constructors cannot be used.

Reply:
I mean State and Behavior of the class is not properly defined.

Reply:
At any given point of time, the state of the object is incomplete.

Reply:
There is no *behaviour* in this class, the purpose of the class is to hold data/config (or state as you will).

Reply:
yes, the same i wanted to convey here. Based on the types, the fields are not relevant. Can use only relevant fields with separate classes?